test/e2e/apimachinery/garbage_collector.go

```diff
@@ -322,23 +331,8 @@ var _ = SIGDescribe("Garbage collector", func() {
 			framework.Failf("Failed to create replication controller: %v", err)
 		}
 		// wait for rc to create some pods
-		if err := wait.Poll(5*time.Second, 30*time.Second, func() (bool, error) {
-			pods, err := podClient.List(ctx, metav1.ListOptions{})
-			if err != nil {
-				return false, fmt.Errorf("failed to list pods: %w", err)
-			}
-			// We intentionally don't wait the number of pods to reach
-			// rc.Spec.Replicas. We want to see if the garbage collector and the
-			// rc manager work properly if the rc is deleted before it reaches
-			// stasis.
```

Comment:
this looks like it was lost in the switch to `waitForReplicas`

Reply:
Ah, thanks, good catch.